src/workflows-app/components/WorkflowInfoBox.ts

```diff
@@ -0,0 +1,147 @@
+import { useCallback, useState } from 'react';
+import { div, h, span } from 'react-hyperscript-helpers';
+import { Link } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { collapseStatus, makeStatusLine } from 'src/components/job-common';
+import { useCancellation, useOnMount } from 'src/libs/react-utils';
+import { makeCompleteDate } from 'src/libs/utils';
+import ViewWorkflowScriptModal from 'src/workflows-app/components/ViewWorkflowScriptModal';
+import { fetchMetadata } from 'src/workflows-app/utils/cromwell-metadata-utils';
+
+import { loadAppUrls } from '../utils/app-utils';
+import { TroubleshootingBox } from './TroubleshootingBox';
+
+export type WorkflowInfoBoxProps = {
+  name: string;
+  namespace: string;
+  submissionId: string;
+  workflowId: string;
+  workspaceId: string;
+  showLogModal: (modalTitle: string, logsArray: string, tesLog: string) => void;
+};
+
+type FetchedWorkflowInfoData = {
+  start: string;
+  end?: string;
+  wdlScript?: string;
+  status: string;
+  workflowLog: string;
+  executionDirectory: string;
+};
+
+export const WorkflowInfoBox: React.FC<WorkflowInfoBoxProps> = (props) => {
+  const signal = useCancellation();
+  const [workflowInfo, setWorkflowInfo] = useState<FetchedWorkflowInfoData | undefined>(undefined);
+
+  const loadWorkflowMetadata = useCallback(
+    async (workflowId: string) => {
+      const { cromwellProxyUrlState } = await loadAppUrls(props.workspaceId, 'cromwellProxyUrlState');
+      const metadata = await fetchMetadata({
+        cromwellProxyUrl: cromwellProxyUrlState.state,
+        workflowId,
+        signal,
+        includeKeys: ['start', 'end', 'submittedFiles', 'status', 'workflowLog', 'workflowRoot'],
+        excludeKeys: ['calls'], // NB: Calls can be very large, so it's important for performance to exclude them from the web request (in this component)
+      });
+      const data: FetchedWorkflowInfoData = {
+        start: metadata.start,
+        end: metadata.end,
+        wdlScript: metadata.submittedFiles.workflow,
+        status: metadata.status,
+        workflowLog: metadata.workflowLog,
+        executionDirectory: metadata.workflowRoot,
+      };
+      setWorkflowInfo(data);
+    },
+    [props.workspaceId, signal]
+  );
+
+  useOnMount(() => {
+    const load = async () => {
+      try {
+        await Promise.resolve(loadWorkflowMetadata(workflowId));
```

Comment:
`Promise.resolve` is unnecessary here. `loadWorkflowMetadata` returns a Promise.

```suggestion
        await loadWorkflowMetadata(workflowId);
```